src/main/java/WordInMatrix.java

```diff
@@ -0,0 +1,133 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class WordInMatrix {
+    public String getCells(String matrix, String word) {
+        checkingForErrors(matrix, word);
+        matrix = matrix.toUpperCase();
+        word = word.toUpperCase();
+        if (word.length() == 1 && matrix.contains(word)) {
+            return addToCellsSequence(matrix.indexOf(word), (int) Math.sqrt(matrix.length()))
+                    .replace("->", "");
+        }
+        for (int i = 0; i < matrix.length() && word.length() > 1; i++) {
+            int letterPosition = matrix.indexOf(word.charAt(0), i);
+            if (letterPosition == -1) {
+                throw new RuntimeException("Your request is not acceptable, try to change word");
+            }
+            List<String> cellsSequence = new ArrayList<>();
+            int row = (int) Math.sqrt(matrix.length());
+            String backup = matrix;
+            matrix = markMyWay(matrix, letterPosition, word.charAt(0));
+            cellsSequence = checkSells(matrix, word.substring(1),
+                    letterPosition, row, cellsSequence);
+            if (!(cellsSequence.get(cellsSequence.size() - 1).equals("Wrong word"))) {
+                StringBuilder letterCells = new StringBuilder();
+                letterCells.append(addToCellsSequence(letterPosition, row));
+                for (String cell : cellsSequence) {
+                    letterCells.append(cell);
+                }
+                return letterCells.toString().replaceFirst("->", "");
+            }
+            matrix = backup;
+        }
+        throw new RuntimeException("Your request is not acceptable. Please, try to change word");
+    }
+
+    private List<String> checkSells(String matrix, String word,
```

Comment:
ideally, it's better to split a logic from this method into several ones, try to keep your methods smaller and it will improve readability